src/main/rules/GCI604/java/GCI604.asciidoc

```diff
@@ -0,0 +1,31 @@
+Spring Retry: Inconsistent configuration
+= Inconsistent Spring Retry Configuration
+
+An inconsistent configuration of Spring Retry can lead to blocking threads and increased energy usage.
+
+With Spring Retry, you can mark operations as retryable using `@Retryable`. However, misconfiguration (e.g., too many retries or long backoff periods) can block threads unnecessarily — consuming CPU cycles and energy.
+
+This rule applies to usages of `@Retryable` only.
+A threshold `max.timeout` defines the maximum acceptable blocking time (in milliseconds).
+
+`max.timeout=50000`
+
+== Noncompliant Code Example
+
+[source,java]
+----
+@Retryable(maxAttempts = 4, backoff = @Backoff(delay = 10, multiplier = 100))
+void retryService(String sql);
+----
+
+In this example, retry delays compound to a total retry time of over 10,000,000 ms (10,000 seconds), which exceeds the `max.timeout` threshold.
+
+== Compliant Solution
+
+[source,java]
+----
+@Retryable(maxAttempts = 5, backoff = @Backoff(delay = 10))
```

Comment:
maybe give some other compliant examples with acceptable "multiplier" or the "maxDelay" given to limit the maximum timeout
because we can have a relevant use case that the multiplier is relevant (for example, call an external URL to limit de spend of too much call tokens) ... but I agree, we have to control it not to consume a lot of resources.